controllers/yawol-cloud-controller/targetcontroller/node_controller_test.go

```diff
@@ -215,6 +220,84 @@ var _ = Describe("Check loadbalancer reconcile", Serial, Ordered, func() {
 				return helper.ErrNoEventFound
 			}, time.Second*5, time.Millisecond*500).Should(Succeed())
 		})
+		It("ignore terminating Node and check", func() {
+			By("create node")
+			node := v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "nodebrk",
+					Namespace: "default"},
+				Spec: v1.NodeSpec{},
+				Status: v1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:               v1.NodeReady,
+							Status:             v1.ConditionTrue,
+							LastHeartbeatTime:  metav1.Time{},
+							LastTransitionTime: metav1.Time{},
+							Reason:             "Ready",
+							Message:            "Ready",
+						},
+						{
+							Type:               NodeTerminationCondition,
+							Status:             v1.ConditionTrue,
+							LastHeartbeatTime:  metav1.Time{},
+							LastTransitionTime: metav1.Time{},
+							Reason:             "Terminating",
+							Message:            "Terminating",
+						},
+					},
+					Addresses: []v1.NodeAddress{
+						{
+							Type:    v1.NodeInternalIP,
+							Address: "10.10.10.42",
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, &node)).Should(Succeed())
+
+			By("check node in LB")
+			assertFirstNode("node1")
```

Comment:
Where does node `node1` come from/get created?

Can't we explicitly assert that the `nodebrk` node is not in the endpoints slice.

Reply:
The surrounding `Describe` is marked as `Ordered`. That is `node1` is the same node as from the previous `It`. Still including `node1` here also sort of checks that a terminating node does not cause other nodes to start missing from the LB.

Reply:
Still a bit implicit, but I'm fine with it now, as `assertLBWithOneEndpoint` is a lot more obvious.